validation/validation_test.go

```diff
@@ -0,0 +1,47 @@
+package validation
+
+import (
+	"testing"
+
+	roles "github.com/hpcloud/fissile/model"
+)
+
+func TestValidateConfigMap(t *testing.T) {
+	newRun := func(memory int, cpu int, name string, externalPort string, internalPort string, protocol string) *roles.RoleRun {
+		return &roles.RoleRun{
+			VirtualCPUs: cpu,
+			Memory:      memory,
+			ExposedPorts: []*roles.RoleRunExposedPort{{Name: name, External: externalPort,
+				Internal: internalPort, Protocol: protocol,
+			}},
+		}
+	}
+
+	var (
+		validRun      = newRun(10, 2, "test", "1", "2", "UDP")
+		wrongProtocol = newRun(10, 2, "test", "1", "2", "AA")
+		wrongPorts    = newRun(10, 2, "test", "0", "-1", "UDP")
+		negativeField = newRun(-10, 2, "test", "1", "2", "UDP")
+	)
+
+	tests := map[string]struct {
+		run     *roles.RoleRun
+		isValid bool
+	}{
+		"nil":            {nil, false},
+		"valid":          {validRun, true},
+		"wrong protocol": {wrongProtocol, false},
+		"wrong ports":    {wrongPorts, false},
+		"negative field": {negativeField, false},
+	}
+
+	for name, tc := range tests {
+		errs := ValidateRoleRun(tc.run)
+		if tc.isValid && len(errs) > 0 {
+			t.Errorf("%v: unexpected error: %v", name, errs)
+		}
+		if !tc.isValid && len(errs) == 0 {
```

Comment:
Ideally, we should be checking that it's returning the correct sorts of errors, too. E.g. if we expect a non-negative field error, but got a invalid integer parse error instead.